Justfile

```diff
@@ -23,52 +26,62 @@ _default:
 
 # ---------------------------------------------------------------------------- #
 
+# Print the paths of all of the binaries
+bins:
+    @echo "{{PKGDB_BIN}}"
+    @echo "{{FLOX_BIN}}"
+
+# ---------------------------------------------------------------------------- #
+
+# Build only pkgdb
 build-pkgdb:
     @make -C pkgdb -j;
 
+# Build only flox
 build-cli: build-pkgdb
-    @pushd cli; cargo build; popd
+    @pushd cli; cargo build -q; popd
 
 # Build the binaries
 build: build-cli
 
 
 # ---------------------------------------------------------------------------- #
 
+# Run the pkgdb tests
 test-pkgdb: build-pkgdb
     @make -C pkgdb -j tests;
     @make -C pkgdb check;
 
 # Run the end-to-end test suite
 functional-tests +bats_args="": build
-    @flox-tests {{bats_args}};
+    @flox-tests --pkgdb "{{PKGDB_BIN}}" --flox "{{FLOX_BIN}}" {{bats_args}}
 
-# Run the integration test suite
-integ-tests: build
-    @flox-cli-tests --pkgdb "${PWD}/pkgdb/bin/pkgdb"        \
-                    --flox "${PWD}/cli/target/debug/flox";
+# Run the CLI integration test suite
+integ-tests +bats_args="": build
+    @flox-cli-tests --pkgdb "{{PKGDB_BIN}}" --flox "{{FLOX_BIN}}" {{bats_args}}
 
-# Run a specific 'bats' test file
-bats-file file: build
-    @flox-tests --tests "{{file}}";
+# Run a specific CLI integration test file by name (not path)
+integ-file file +bats_args="": build
+    @flox-cli-tests --tests "{{file}}" --pkgdb "{{PKGDB_BIN}}" \
+     --flox "{{FLOX_BIN}}" {{bats_args}}
 
-# Run the Rust unit tests
+# Run the CLI unit tests
 unit-tests regex="": build
     @pushd cli;                            \
      {{cargo_test_invocation}} {{regex}};  \
      popd;
 
-# Run the test suite, including impure tests
+# Run the CLI unit tests, including impure tests
 impure-tests regex="": build
     @pushd cli;                                                     \
      {{cargo_test_invocation}} {{regex}} --features "extra-tests";  \
      popd;
 
-# Run the entire test suite, not including impure tests
-test-cli: build unit-tests functional-tests integ-tests
+# Run the entire CLI test suite
+test-cli: impure-tests integ-tests functional-tests
```

Comment:
What's the reason to remove build and add impure-tests?

Doesn't the test suite still depend on building prior to running the tests?

Reply:
Build is already included in those targets